File: WebModule/common/models/Invoiceline.php
<?php

namespace common\models;

use Yii;

/**
 * This is the model class for table "invoice_lines".
 *
 * @property int $id
 * @property int $item_id
 * @property float $qty
 * @property float $total
 * @property int $invoice_id
 *
 * @property Invoice $invoice
 * @property Item $item
 */
class InvoiceLine extends \yii\db\ActiveRecord
{
    /**
     * {@inheritdoc}
     */
    public static function tableName()
    {
        return 'invoice_lines';
    }

    /**
     * {@inheritdoc}
     */
    public function rules()
    {
        return [
            [['item_id', 'qty', 'total', 'invoice_id'], 'required'],
            [['item_id', 'invoice_id'], 'integer'],
            [['qty', 'total'], 'number'],
            [['invoice_id'], 'exist', 'skipOnError' => true, 'targetClass' => Invoice::class, 'targetAttribute' => ['invoice_id' => 'id']],
            [['item_id'], 'exist', 'skipOnError' => true, 'targetClass' => Item::class, 'targetAttribute' => ['item_id' => 'id']],
        ];
    }

    /**
     * {@inheritdoc}
     */
    public function attributeLabels()
    {
        return [
            'id' => 'ID',
            'item_id' => 'Item ID',
            'qty' => 'Qty',
            'total' => 'Total',
            'invoice_id' => 'Invoice ID',
        ];
    }

    /**
     * Gets query for [[Invoice]].
     *
     * @return \yii\db\ActiveQuery
     */
    public function getInvoice()
    {
        return $this->hasOne(Invoice::class, ['id' => 'invoice_id']);
    }

    /**
     * Gets query for [[Item]].
     *
     * @return \yii\db\ActiveQuery
     */
    public function getItem()
    {
        return $this->hasOne(Item::class, ['id' => 'item_id']);
    }

    public function fields() {
        $fields = parent::fields();

        unset($fields['item_id'], $fields['invoice_id']);

        $fields['item'] = function() {
            $item = $this->getItem()->one();
            return $item ? $item : null;
        };

        $fields['invoice'] = function() {
            $invoice = $this->getInvoice()->one();
            return $invoice ? $invoice : null;
        };

        return $fields;
    }
}

File: Mobile/app/src/main/java/com/example/gofuel/model/invoice/InvoiceLine.java
package com.example.gofuel.model.invoice;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.example.gofuel.model.item.Item;
import com.example.gofuel.model.pump.Pump;

@Entity(tableName = "invoicelines")
public class InvoiceLine {
    @PrimaryKey
    private final int id;
    private Item item;
    private int qty;
    private double total;
    private Invoice invoice;

    public InvoiceLine(int id, Item item, int qty, double total, Invoice invoice) {
        this.id = id;
        this.item = item;
        this.qty = qty;
        this.total = total;
        this.invoice = invoice;
    }

    public int getId() {
        return id;
    }

    public Item getItem() {
        return item;
    }

    public int getQty() {
        return qty;
    }

    public double getTotal() {
        return total;
    }

    public Invoice getInvoice() {
        return invoice;
    }

    public void setItem(Item item) {
        this.item = item;
    }

    public void setQty(int qty) {
        this.qty = qty;
    }

    public void setTotal(double total) {
        this.total = total;
    }

    public void setInvoice(Invoice invoice) {
        this.invoice = invoice;
    }
}
